web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx

```diff
@@ -0,0 +1,176 @@
+import { Balance, Button, Input } from '@massalabs/react-ui-kit';
+import { FiUsers } from 'react-icons/fi';
+import useResource from '../../../custom/api/useResource';
+import { AccountObject } from '../../../models/AccountModel';
+import { useLocation } from 'react-router-dom';
+import { parseForm } from '../../../utils/parseForm';
+import { useState } from 'react';
+
+// remove padding from Balance component
+
+interface IErrorObject {
+  amount?: string;
+  fees?: string;
+  recipient?: string;
+  error?: string;
+}
+
+function Send() {
+  const { state } = useLocation();
+  const nickname = state.nickname;
+  const { data: accounts = [] } = useResource<AccountObject[]>('accounts');
+  const account = accounts.find((account) => account.nickname === nickname);
+  const balance = parseInt(account!.candidateBalance);
+  const [amount, setAmount] = useState<number>(0);
+  const [error, setError] = useState<IErrorObject | null>(null);
+  const [recipient, setRecipient] = useState('');
+  const selectPercentClass = 'mr-3.5 hover:cursor-pointer';
+
+  // refactor to switch => todo
+  // Check to see if address exists in our system
+
+  function validate(e: any) {
+    const formObject = parseForm(e);
+    const { amount, recipient } = formObject;
+
+    if (!amount) {
+      setError({ amount: 'Amount is required' });
+      return false;
+    }
+    const int = parseInt(amount);
+    if (Number.isNaN(int)) {
+      setError({ amount: 'Please enter a valid number' });
+      return false;
+    }
+    if (!recipient) {
+      setError({ recipient: 'Recipient is required' });
+      return false;
+    }
+    setError(null);
+    return true;
+  }
+
+  // This function is going to be used for the backend
+  function handleSubmit(e: any) {
+    e.preventDefault();
+    if (!validate(e)) return;
+
+    const formObject = parseForm(e);
+
+    const { amount, recipient } = formObject;
+
+    // console logs to avoid build fail
+    console.log(amount);
+    console.log(recipient);
+    console.log('submit');
+  }
+  function formatStandard(num: number) {
+    return num.toLocaleString('en-US', {
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    });
+  }
```

Comment:
We must a SINGLE source of truth for that placed under **util** for example. This will be used everywhere. So let's import and use but the function remains the same for all.

Reply:
agree !

Reply:
I created a dedicated tsx un /utils dir ! hope that's fine